test/project_types/script/script_project_test.rb

```diff
@@ -9,7 +9,39 @@ def setup
       @script_name = 'name'
     end
 
+    def test_create_when_directory_does_not_exist
+      Dir
+        .expects(:exist?)
+        .with(@script_name)
+        .returns(false)
```

Comment:
Hum, interesting.  Why is `Dir.exists?` not moved to `Context`?  It feels like `Context` is a big wrapper around all file system operations, right?

Reply:
Good catch!